spyql/interactive.py

```diff
@@ -0,0 +1,97 @@
+import os
+
+from .cli import clean_query, parse
+from .processor import Processor
+from .log import *
+
+class Q:
+  def __init__(self, query: str) -> None:
+    """
+    Make spyql interactive.
+
+    [ IMPORT python_module [ AS identifier ] [, ...] ]
+    SELECT [ DISTINCT | PARTIALS ]
+        [ * | python_expression [ AS output_column_name ] [, ...] ]
+        [ FROM csv | spy | text | python_expression | json [ EXPLODE path ] ]
+        [ WHERE python_expression ]
+        [ GROUP BY output_column_number | python_expression  [, ...] ]
+        [ ORDER BY output_column_number | python_expression
+            [ ASC | DESC ] [ NULLS { FIRST | LAST } ] [, ...] ]
+        [ LIMIT row_count ]
+        [ OFFSET num_rows_to_skip ]
+        [ TO csv | json | spy | sql | pretty | plot ]
+
+    Usage
+    -----
+
+    .. code-block:: python
+
+      >>> q = Q("IMPORT numpy SELECT numpy.mean(data->salary) FROM data WHERE data->name == 'akash'")
+      >>> q(data = data)
+
+    Args
+    ----
+
+      query(str): SpyQL string
+      input_opt/output_opt: kwargs for the input and writers, in this case of interactive mode we can
+        ignore these
+    """
+    self.query = query
+    self.parsed, self.strings = parse(clean_query(query))
+    self.output_path = None
+
+    # from logic has to be improved to include the files, python datastructures
+    _from = self.parsed["from"]
+    input_options = {}
+    interactive = False
+    if _from == None:
+      # SELECT 1
+      pass
+    elif os.path.exists(_from):
+      # SELECT * FROM /tmp/spyql.jsonl
+
+      ext = _from.split(".")[-1].lower()
+      processor_from = {"jsonl": "JSON", "csv": "CSV"}.get(ext, None)
+      if processor_from == None:
+        user_warning(f"Unsupported file extension: '{ext}', loading as 'TEXT'")
+        processor_from = "TEXT"
+
+      self.parsed["from"] = processor_from
+      interactive = True
+      input_options = {"filepath": _from}
+    else:
+      # SELECT * FROM data
+      self.parsed["interactive"] = True
+      interactive = True
+      input_options = {"source": _from}
+
+    _to = self.parsed["to"]
+    if _to == None:
+      self.parsed["to"] = "PYTHON" # force return to python
```

Comment:
The values here are more about the structure of the output, whereas in this case it is telling more about the output location (bad word?), which is the caller. But I get the idea that user can get output in multiple format, I think I'll implement it in the `Output` object which is returned by the `go`. The `Output` can have functions like `to("dict")` or dedicated methods `to_list, to_records, ...`.

When I work on the output I'll implement as `to()`, will change this before final merge.

Reply:
`"MEMORY"` makes sense! For context, spyql once had a `"PY"` format that was similar to the `"SPY"` format but in plain text. 
Yes, let's go with the `to` approach.
What do you think about leaving the result object implementation as well as support  to follow-up PRs? In this PR we would try to set the foundation for interactive spyql and then we would add new features. The idea is simply to don't make this PR too long, but if you prefer we can continue iterating on this one.

Reply:
Sure. I think for this version you can keep the `ProcessedOutput` object since there is no change in CLI behaviour except a minor `spyql.log.user_info(f"Output Meta: {out}")` added at the end of `main()`.